pkg/apis/operator/v1alpha1/apiserver_types.go

```diff
@@ -26,12 +26,21 @@ type APIServerSpec struct {
 	// TLSSecretRef references the TLS certificate and private key for serving the KubeCarrier API.
 	// +optional
 	TLSSecretRef *ObjectReference `json:"tlsSecretRef,omitempty"`
+	// +optional
+	// Authentication configuration
+	Authentication *Authentication `json:"authentication,omitempty"`
```

Comment:
This should not be optional I think? Otherwise what authentication mode will be enabled if it is omitted? (Anonymous should be enabled as default I think)

Reply:
I current implementation we have everything optional and we are adding `Anonymous` and `Token` providers by default.
I can add `Anonymous` provider to auth modes in `Order` is empty here
https://github.com/kubermatic/kubecarrier/blob/7c27999632942978b0f471738d8c6760098000e7/pkg/internal/resources/apiserver/apiserver.go#L223
what do you think?

Reply:
> This should not be optional I think? Otherwise what authentication mode will be enabled if it is omitted? (Anonymous should be enabled as default I think)

If the things is optional the defaulting webhook should fill the defaults in.